### DISCUSSION TITLE
ERROR while Running

### DISCUSSION THREAD

**Discussion by Howpow88** [Valdes-Tresanco-MS/gmx_MMPBSA#388] [General]
Please help with the following Error 
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file

How to fix PBD for the comman run?

**Comment by Valdes-Tresanco-MS** [comment]
Please, provide more info about your system and the calculation. I can't identify the error with just that warning

**Reply by Howpow88** [reply]
This is the full system information
 
(gmxMMPBSA) [u055769@slhlnxws065 mmpbsa]$ gmx_MMPBSA -O -i mmpbsa.in -cs md_0_1.tpr -ct md_0_2.xtc -ci index.ndx -cg 3 4 -cp topol.top -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv
[INFO   ] Starting gmx_MMPBSA v1.5.0.3
[INFO   ] Command-line
  gmx_MMPBSA -O -i mmpbsa.in -cs md_0_1.tpr -ct md_0_2.xtc -ci index.ndx -cg 3 4 -cp topol.top -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv

[INFO   ] Checking external programs...
[INFO   ] cpptraj found! Using /home/u055769/miniconda3/envs/gmxMMPBSA/bin/cpptraj
[INFO   ] tleap found! Using /home/u055769/miniconda3/envs/gmxMMPBSA/bin/tleap
[INFO   ] parmchk2 found! Using /home/u055769/miniconda3/envs/gmxMMPBSA/bin/parmchk2
[INFO   ] sander found! Using /home/u055769/miniconda3/envs/gmxMMPBSA/bin/sander
[INFO   ] Using GROMACS version > 5.x.x!
[INFO   ] gmx found! Using /home/u055769/miniconda3/envs/gmxMMPBSA/bin.AVX2_256/gmx
[INFO   ] Checking external programs...Done.

[INFO   ] Building AMBER topologies from GROMACS files...
[INFO   ] Get PDB files from GROMACS structures files...
[INFO   ] Making gmx_MMPBSA index for complex...
[INFO   ] Normal Complex: Saving group 3_4 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_COM.pdb
[INFO   ] No receptor structure file was defined. Using ST approach...
[INFO   ] Using receptor structure from complex to generate AMBER topology
[INFO   ] Normal Receptor: Saving group 3 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_REC.pdb
[INFO   ] No ligand structure file was defined. Using ST approach...
[INFO   ] Using ligand structure from complex to generate AMBER topology
[INFO   ] Normal Ligand: Saving group 4 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_LIG.pdb
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
[WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file
  File "/home/u055769/miniconda3/envs/gmxMMPBSA/bin/gmx_MMPBSA", line 10, in <module>
    sys.exit(gmxmmpbsa())
  File "/home/u055769/miniconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/app.py", line 101, in gmxmmpbsa
    app.make_prmtops()
  File "/home/u055769/miniconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/main.py", line 533, in make_prmtops
    self.FILES.mutant_receptor_prmtop, self.FILES.mutant_ligand_prmtop) = maketop.buildTopology()
  File "/home/u055769/miniconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 116, in buildTopology
    self.gmx2pdb()
  File "/home/u055769/miniconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 390, in gmx2pdb
    self.resi, self.resl, self.orderl = res2map(self.indexes, self.complex_str)
  File "/home/u055769/miniconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/utils.py", line 505, in res2map
    for e in value['num']:
TypeError: string indices must be integers
Exiting. All files have been retained.

**Reply by Valdes-Tresanco-MS** [reply]
It looks like your topology is inconsistent or has elements not supported by ParmEd. Please, send your files (*.top + *.itp + folder if applied, *.tpr, *.ndx, 5frames *.xtc and the groups number)

**Reply by Howpow88** [reply]
how do i get these files to you as it says files not supported when i drop them in the chat

**Reply by Valdes-Tresanco-MS** [reply]
Create a zip file...

**Comment by Howpow88** [comment]
[md_fit.zip](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/files/11907928/md_fit.zip)

**Reply by Valdes-Tresanco-MS** [reply]
Please, attach all required files. This zip only contain the topology and the tpr

**Comment by Howpow88** [comment]
[md_0_3.zip](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/files/11908122/md_0_3.zip)

**Comment by Howpow88** [comment]
here is the xtc which other files do you need?

**Comment by Howpow88** [comment]
i don't have a itp file?

**Comment by Howpow88** [comment]
do you mean an index file?

**Comment by Howpow88** [comment]
[mmpbsa.zip](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/files/11908156/mmpbsa.zip)
this is my .in file

**Comment by Howpow88** [comment]
my ligand is the first 0-1689 atoms the rest is the receptor

**Reply by Valdes-Tresanco-MS** [reply]
According to the autogenerated index and the structure generated by it, the ligand chain has 3368 atoms, which does not coincide with the number you report.
There are also residue names that are not common, such as HISH. ParmEd interprets this as a modified residue (such as phosphoserine, for example), so it assigns the HETATM tag. There is a discontinuity in the numbers of the ligand residues, so multiple chain IDs are generated. These inconsistencies can affect the performance of the program and its dependencies.

**Comment by Howpow88** [comment]
i'm using this command gmx_MMPBSA -O -i mmpbsa.in -cs com.tpr -ct com_traj.xtc -ci index.ndx -cg 3 4 -cp topol.top -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv
[index.zip](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/files/11908275/index.zip)

**Reply by Valdes-Tresanco-MS** [reply]
For some reason, I can't process this topology.
```python
Traceback (most recent call last):
  File "/home/mario/PycharmProjects/gmx_MMPBSA/scratch_2.py", line 4, in <module>
    top = parmed.load_file('/home/mario/Documents/gmx_MMPBSA_tests/1.6.x/users/issue388/topol.top')
  File "/home/mario/PycharmProjects/ParmEd/parmed/formats/registry.py", line 213, in load_file
    return cls(filename, *args, **kwargs)
  File "/home/mario/PycharmProjects/ParmEd/parmed/gromacs/gromacstop.py", line 251, in __init__
    self.read(fname, defines, parametrize)
  File "/home/mario/PycharmProjects/ParmEd/parmed/gromacs/gromacstop.py", line 314, in read
    bond, bond_type = self._parse_bonds(line, bond_types, molecule.atoms)
  File "/home/mario/PycharmProjects/ParmEd/parmed/gromacs/gromacstop.py", line 511, in _parse_bonds
    bond = Bond(atoms[i], atoms[j])
IndexError: list index out of range
```

**Comment by huaiaee** [comment]
> Please help with the following Error [WARNING] Could not find the head atom of the next template! Bond pattern may be wrong, which could lead to extra TER cards in a PDB file

I also encountered the same problem. Have you finally solved it?

**Reply by marioernestovaldes** [reply]
can you send the files you are using?
